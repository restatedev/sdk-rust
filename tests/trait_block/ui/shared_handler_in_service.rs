use restate_sdk::prelude::*;

#[restate_sdk::service]
trait SharedHandlerInService {
    #[shared]
    async fn my_handler() -> HandlerResult<()>;
}

struct SharedHandlerInServiceImpl;

impl SharedHandlerInService for SharedHandlerInServiceImpl {
    async fn my_handler(&self, _: Context<'_>) -> HandlerResult<()> {
        Ok(())
    }
}

#[tokio::main]
async fn main() {
    tracing_subscriber::fmt::init();
    HttpServer::new(
        Endpoint::builder()
            .with_service(SharedHandlerInServiceImpl.serve())
            .build(),
    )
        .listen_and_serve("0.0.0.0:9080".parse().unwrap())
        .await;
}
