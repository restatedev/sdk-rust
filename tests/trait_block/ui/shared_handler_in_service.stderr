error: Service handlers cannot be annotated with #[shared]
 --> tests/ui/shared_handler_in_service.rs:6:14
  |
6 |     async fn my_handler() -> HandlerResult<()>;
  |              ^^^^^^^^^^

error[E0405]: cannot find trait `SharedHandlerInService` in this scope
  --> tests/ui/shared_handler_in_service.rs:11:6
   |
11 | impl SharedHandlerInService for SharedHandlerInServiceImpl {
   |      ^^^^^^^^^^^^^^^^^^^^^^ not found in this scope
